src/main/groovy/org/mockito/release/internal/gradle/util/GitPluginUtil.java

```diff
@@ -0,0 +1,27 @@
+package org.mockito.release.internal.gradle.util;
+
+import org.gradle.api.Action;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.mockito.release.internal.gradle.GitCommitTask;
+import org.mockito.release.internal.gradle.GitPlugin;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * Utility methods for {@link org.mockito.release.internal.gradle.GitPlugin}
+ */
+public class GitPluginUtil {
+
+    public static void registerChangesIfGitPluginApplied(final Project project, final List<File> changedFiles,
```

Comment:
Great idea to push that complexity to separate method!

Feedback:
1. Consider moving this method directly to GitPlugin class because it is coupled with that plugin anyway. This would result in simpler code I think.
2. when you pass task instance to the method, you automatically have access to Gradle project object: ```task.getProject()```